rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -1003,9 +1003,10 @@ void FleetUpdateHandle::add_robot(
     command = std::move(command),
     start = std::move(start),
     handle_cb = std::move(handle_cb),
-    fleet = shared_from_this()](
+    fleet_wptr = weak_from_this()](
       rmf_traffic::schedule::Participant participant)
     {
+      auto fleet = fleet_wptr.lock();
```

Comment:
Whenever we lock a `std::weak_ptr`, we always need to check if the locking was successful and simply return right away if it wasn't, i.e.:

```
if (!fleet)
  return;
```

Reply:
73da754